handlers/_swagger.js

```diff
@@ -319,31 +319,32 @@ function registerRoute(app, auth, additionalMiddleware, method, path, data, allo
             setDefaultQueryParams(req, data, logger);
             setDefaultHeaders(req, data, logger);
 
-            //Wrap the set function, which is responsible for setting headers
-            if (allowedTypes) {
-                //Validate that the content-type is correct per the swagger definition
-                wrapCall(res, 'set', function (name, value) {
-                    if (name === 'Content-Type') {
-                        var type = value.split(';')[0]; //parse off the optional encoding
-                        if (!_.contains(allowedTypes, type)) {
-                            logger.warn('Invalid content type specified: %s. Expecting one of %s', type, allowedTypes);
-                        }
+            // Wrap the set function, which is responsible for setting headers
+            var type = '';
+            // Validate that the content-type is correct per the swagger definition
+            wrapCall(res, 'set', function (name, value) {
+                if (name === 'Content-Type') {
+                    type = value.split(';')[0]; //parse off the optional encoding
+                    if (allowedTypes && !_.contains(allowedTypes, type)) {
+                        logger.warn('Invalid content type specified: %s. Expecting one of %s', type, allowedTypes);
                     }
-                });
-            }
+                }
+            });
 
             if (responseModelValidationLevel) {
                 var responseSender = res.send;
                 res.send = function (body) {
                     var isBodyValid = isValidDataType(body);
                     if (!isBodyValid) {
-                        try { //body can come in as JSON, we want it unJSONified
-                            body = JSON.parse(body);
-                        } catch (err) {
-                            logger.error('Unexpected format when attempting to validate response');
-                            res.send = responseSender;
-                            responseSender.call(res, body);
-                            return;
+                        if ( type === 'application/json') {
```

Comment:
There's an extra space between the bracket and `type` ...

Also, how does this affect users who are not explicitly setting their response content type?
Has express already set that header by this time?
Above, should the default value for `type` be `'application/json'`?
Or should this check be more like: `if (!type || type === 'application/json')`?